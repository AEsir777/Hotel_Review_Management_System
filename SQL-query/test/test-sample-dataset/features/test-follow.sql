/**** add friend ******/
/*** Test insert and isFriend ******/
-- check
CALL GetFollowers('U3'); -- the result shouldn't include U8
CALL GetFollowers('U8'); -- the result shouldn't include U3
CALL GetFollowings('U3'); -- the result shouldn't include U8
CALL GetFollowings('U8'); -- the result shouldn't include U3
-- success case
CALL InsertFriend('U3', 'U8');
CALL InsertFriend('U8', 'U3');
-- -- check
CALL GetFollowers('U3'); -- the result now include U8
CALL GetFollowers('U8'); -- the result now include U3
CALL GetFollowings('U3'); -- the result now include U8
CALL GetFollowings('U8'); -- the result now include U3

-- check: exist before
CALL GetFollowers('U1');
CALL GetFollowers('U2');
CALL GetFollowings('U1');
CALL GetFollowings('U2');


-- check: insert fail
CALL InsertFriend('U1', 'U2');
CALL InsertFriend('U1', 'U200');