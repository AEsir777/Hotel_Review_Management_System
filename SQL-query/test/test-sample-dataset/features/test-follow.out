/**** add friend ******/
/*** Test insert and isFriend ******/
// -- check
CALL GetFollowers('U3');
output: 
U1	User One
U2	User Two
U5	User Five
expected output:
U1	User One
U2	User Two
U5	User Five
// the result shouldn't include U8

CALL GetFollowers('U8'); 
output: 
U6	User Six
U7	User Seven
expected output:
U6	User Six
U7	User Seven
// the result shouldn't include U3

CALL GetFollowings('U3');
output: 
U1	User One
U2	User Two
U5	User Five
expected output:
U1	User One
U2	User Two
U5	User Five
// the result shouldn't include U8

CALL GetFollowings('U8');
output: 
U7	User Seven
expected output:
U7	User Seven
// the result shouldn't include U3


CALL InsertFriend('U3', 'U8');
CALL InsertFriend('U8', 'U3');
2 row(s) affected   
// expected output: succesfully inserted into the row

// -- check
CALL GetFollowers('U3');
output: 
U1	User One
U2	User Two
U5	User Five
U8	User Eight
expected output:
U1	User One
U2	User Two
U5	User Five
U8	User Eight
// the result should include U8

CALL GetFollowers('U8'); 
output: 
U3	User Three
U6	User Six
U7	User Seven
expected output:
U3	User Three
U6	User Six
U7	User Seven
// the result should include U3

CALL GetFollowings('U3');
output: 
U1	User One
U2	User Two
U5	User Five
U8	User Eight
expected output:
U1	User One
U2	User Two
U5	User Five
U8	User Eight
// the result should include U8

CALL GetFollowings('U8');
output: 
U3	User Three
U7	User Seven
expected output:
U3	User Three
U7	User Seven
// the result should include U3

-- check: exist before
CALL GetFollowers('U1');
output: 
U2	User Two
U3	User Three
U4	User Four
U5	User Five
expected output:
U2	User Two
U3	User Three
U4	User Four
U5	User Five

CALL GetFollowers('U2');
output: 
U1	User One
U3	User Three
U4	User Four
expected output:
U1	User One
U3	User Three
U4	User Four

CALL GetFollowings('U1');
output: 
U2	User Two
U3	User Three
U4	User Four
U5	User Five
expected output:
U2	User Two
U3	User Three
U4	User Four
U5	User Five

CALL GetFollowings('U2');
output: 
U1	User One
U3	User Three
U4	User Four
expected output:
U1	User One
U3	User Three
U4	User Four

-- check: insert fail
// CALL InsertFriend('U1', 'U2');
Error Code: 1062. Duplicate entry 'U1-U2' for key 'friend.PRIMARY'
// expected output: fail

// CALL InsertFriend('U1', 'U200');
Error Code: 1452. Cannot add or update a child row: a foreign key constraint fails (`cs348`.`friend`, CONSTRAINT `friend_ibfk_2` FOREIGN KEY (`uid2`) REFERENCES `userfile` (`uid`) ON DELETE CASCADE)
// expected output: fail
